tools/mo/openvino/tools/mo/main.py

```diff
@@ -308,24 +312,50 @@ def raise_ie_not_found():
         from openvino.tools.mo.front.onnx.register_custom_ops import get_front_classes
         import_extensions.load_dirs(argv.framework, extensions, get_front_classes)
 
-    return argv
 
+def check_fallback(argv : argparse.Namespace):
+    fallback_reasons = {}
 
-def prepare_ir(argv):
-    argv = arguments_post_parsing(argv)
+    if not any(deduce_framework_by_namespace(argv)): # no legacy path
+        return fallback_reasons
+
+    if hasattr(argv, 'use_new_frontend') or hasattr(argv, 'use_legacy_frontend'): # frontend chosen explicitly
+        return fallback_reasons
+
+    fallback_reasons['extensions'] = \
+        lambda argv : hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
+            and argv.extensions != import_extensions.default_path() # extensions arg has default value
+    fallback_reasons['transformations_config'] = \
+        lambda argv: hasattr(argv, 'transformations_config') and argv.transformations_config is not None and len(argv.transformations_config) > 0
 
+    reasons = [reason for reason, is_applicable in fallback_reasons.items() if is_applicable(argv)]
+    return reasons
+
+
+def prepare_ir(argv : argparse.Namespace):
+    argv = arguments_post_parsing(argv)
     t = tm.Telemetry()
     graph = None
     ngraph_function = None
     moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
-
     if moc_front_end:
-        t.send_event("mo", "conversion_method", moc_front_end.get_name() + "_frontend")
-        moc_front_end.add_extension(TelemetryExtension("mo", t.send_event, t.send_error, t.send_stack_trace))
-        ngraph_function = moc_pipeline(argv, moc_front_end)
-    else:
-        t.send_event("mo", "conversion_method", "mo_legacy")
-        graph = unified_pipeline(argv)
+        fallback_reasons = check_fallback(argv)
+        if len(fallback_reasons) == 0:
+            t.send_event("mo", "conversion_method", moc_front_end.get_name() + "_frontend")
+            moc_front_end.add_extension(TelemetryExtension("mo", t.send_event, t.send_error, t.send_stack_trace))
+            ngraph_function = moc_pipeline(argv, moc_front_end)
+            return graph, ngraph_function
+        else: # apply fallback
+            reasons_message = ", ".join(fallback_reasons)
+            argv.use_legacy_frontend = True
+            load_extensions(argv)
+            log.warning("The IR preparation was executed by the legacy MO path. "
+                        "This is a fallback scenario applicable only for some specific cases. "
+                       f"The detailed reason why fallback was executed: not supported {reasons_message} were used. "
+                        "You can specify --use_legacy_frontend flag to force using the legacy MO path to avoid additional checks.")
```

Comment:
Mentioning only `--use_legacy_frontend` in case when the legacy FE has been already triggered automatically seems to be not very relevant because if `--use_legacy_frontend` is used in this case it will produce IR in the same way as without using this key except this warning message won't be printed out. Instead, mentioning `--use_new_frontend` is more useful for the customer because it forces the new behavior and shouldn't involve any logic to decide which frontend is called. I think we should provide a link to a more detailed description of what's going on with all these `use_new_frontend`/`use_legacy_frontend`.

Reply:
done